src/view3d.cpp

```diff
@@ -185,31 +212,132 @@ void View3D::centerOnGeometry(const QModelIndex& index)
+        default:                        status="INCOMPLETE";    break;
+    }
+
+    tfm::printfln("GL_FRAMEBUFFER_%s - %s:%i", status, file, line);
+}
+
+#define glFrameBufferStatus(TARGET) _glFrameBufferStatus(TARGET, __FILE__,__LINE__)
+
+#endif //GL_CHECK_RENDER
+
 
 void View3D::initializeGL()
 {
+    // GLEW has a problem with core contexts. It calls glGetString(GL_EXTENSIONS)​,
+    // which causes GL_INVALID_ENUM on GL 3.2+ core context as soon as glewInit()​ is called.
+    // It also doesn't fetch the function pointers. The solution is for GLEW to use glGetStringi​ instead.
+    // The current version of GLEW is 1.10.0 but they still haven't corrected it.
+    // The only fix is to use glewExperimental​ for now.
+    // https://www.opengl.org/wiki/OpenGL_Loading_Library
+    glewExperimental = true;
     if (glewInit() != GLEW_OK)
     {
         g_logger.error("%s", "Failed to initialize GLEW");
         m_badOpenGL = true;
         return;
     }
+    else
+    {
+        g_logger.info("%s", "Initialized GLEW");
+    }
     g_logger.info("OpenGL implementation:\n"
-                  "GL_VENDOR   = %s\n"
-                  "GL_RENDERER = %s\n"
-                  "GL_VERSION  = %s",
+                  "GL_VENDOR    = %s\n"
+                  "GL_RENDERER  = %s\n"
+                  "GL_VERSION   = %s\n"
+                  "GLSL_VERSION = %s\n"
+                  "GLEW_VERSION = %s\n",
                   (const char*)glGetString(GL_VENDOR),
                   (const char*)glGetString(GL_RENDERER),
-                  (const char*)glGetString(GL_VERSION));
-    m_shaderProgram->setContext(context());
-    m_meshFaceShader.reset(new ShaderProgram(context()));
-    m_meshFaceShader->setShaderFromSourceFile("shaders:meshface.glsl");
-    m_meshEdgeShader.reset(new ShaderProgram(context()));
-    m_meshEdgeShader->setShaderFromSourceFile("shaders:meshedge.glsl");
+                  (const char*)glGetString(GL_VERSION),
+                  (const char*)glGetString(GL_SHADING_LANGUAGE_VERSION),
+                  (const char*)glewGetString(GLEW_VERSION));
+
+#ifdef GL_CHECK_RENDER
+    glCheckError();
+#endif
+
+    initCursor(10, 1);
+    initAxes();
+    initGrid(2.0f);
+
+#ifdef GL_CHECK_RENDER
+    //glFrameBufferStatus(m_incrementalFramebuffer->handle());
+    glCheckError();
+#endif
+
+    m_boundingBoxShader.reset(new ShaderProgram());
+    bool bbox_shader_init = m_boundingBoxShader->setShaderFromSourceFile("shaders:bounding_box.glsl");
+
+    m_meshFaceShader.reset(new ShaderProgram());
+    bool meshface_shader_init = m_meshFaceShader->setShaderFromSourceFile("shaders:meshface.glsl");
+
+    m_meshEdgeShader.reset(new ShaderProgram());
+    bool meshedge_shader_init = m_meshEdgeShader->setShaderFromSourceFile("shaders:meshedge.glsl");
+
     m_incrementalFramebuffer = allocIncrementalFramebuffer(width(), height());
+
+#ifdef GL_CHECK_RENDER
+    //glFrameBufferStatus(m_incrementalFramebuffer->handle());
+    glCheckError();
+#endif
```

Comment:
Ok, I moved this into `glutil.h` and removed those `#ifdef` checks, but to actually enable glCheckError() calls, you still have to `#define GL_CHECK` for now.